src/CMakeLists.txt

```diff
@@ -1,5 +1,165 @@
 ##
-# Lists of source files. Keep ALL lists sorted in alphabetical order.
+# Set up additional backend options for Halide
+##
+option(TARGET_OPENCL "Include OpenCL-C target" ON)
+option(TARGET_OPENGL "Include OpenGL/GLSL target" ON)
+option(TARGET_METAL "Include Metal target" ON)
+option(TARGET_D3D12COMPUTE "Include Direct3D 12 Compute target" ON)
+
+##
+# List of optional targets
+##
+
+set(OPTIONAL_TARGETS
+    AARCH64
+    AMDGPU
+    ARM
+    HEXAGON
+    MIPS
+    NVPTX
+    POWERPC
+    RISCV
+    WEBASSEMBLY
+    X86
+    OPENCL
+    OPENGL
+    METAL
+    D3D12COMPUTE
+    )
+
+##
+# Lists of target-specific source files.
+# Keep ALL lists sorted in alphabetical order.
+##
+
+##
+# Header files
+##
+
+# The externally-visible header files that go into making Halide.h.
+# Don't include anything here that includes llvm headers.
+
+set(AARCH64_HEADER_FILES
+    )
+
+set(AMDGPU_HEADER_FILES
+    )
+
+set(ARM_HEADER_FILES
+    CodeGen_ARM.h
+    )
+
+set(HEXAGON_HEADER_FILES
+    HexagonOffload.h
+    HexagonOptimize.h
+    )
+
+set(MIPS_HEADER_FILES
+    CodeGen_MIPS.h
+    )
+
+set(NVPTX_HEADER_FILES
+    CodeGen_PTX_Dev.h
+    )
+
+set(POWERPC_HEADER_FILES
+    CodeGen_PowerPC.h
+    )
+
+set(RISCV_HEADER_FILES
+    CodeGen_RISCV.h
+    )
+
+set(WEBASSEMBLY_HEADER_FILES
+    CodeGen_WebAssembly.h
+    )
+
+set(X86_HEADER_FILES
+    CodeGen_X86.h
+    )
+
+set(OPENCL_HEADER_FILES
+    CodeGen_OpenCL_Dev.h
+    )
+
+set(OPENGL_HEADER_FILES
+    CodeGen_OpenGL_Dev.h
+    CodeGen_OpenGLCompute_Dev.h
+    InjectOpenGLIntrinsics.h
+    )
+
+set(METAL_HEADER_FILES
+    CodeGen_Metal_Dev.h
+    )
+
+set(D3D12COMPUTE_HEADER_FILES
+    CodeGen_D3D12Compute_Dev.h
+    )
+
+##
+# Source files
+##
+set(AARCH64_SOURCE_FILES
+    )
+
+set(AMDGPU_SOURCE_FILES
+    )
+
```

Comment:
No need to have this span multiple lines. Better to show that it's meant to be empty via `set(AARCH64_SOURCE_FILES "")`